core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -152,8 +152,13 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
       sc.sparkUser,
       sc.appName,
       sc.conf,
-      sc.conf.getOption("spark.mesos.driver.webui.url").orElse(sc.ui.map(_.appUIAddress))
+      sc.conf.getOption("spark.mesos.driver.webui.url").orElse(sc.ui.map(_.appUIAddress)),
+      None,
+      None,
+      sc.conf.getOption("spark.mesos.driver.frameworkId")
     )
+
+    unsetFrameworkID(sc)
```

Comment:
> If no id is specified what will happen?

Mesos generates a unique framework ID when none is specified.  This PR only sets the framework ID on the first spark context created by the driver.

> What if you could use a unique id as aframeworkId plus an increment, so the scheduler can detect all frameworks even the ones coming from multiple contexts? 

We can't guarantee no collisions with that approach.  It would also be difficult for the dispatcher to determine how many frameworks have been started.